### ISSUE TITLE
Vnds fails to start on macOS

### ISSUE THREAD

**Issue by lochlainn** [ajusa/VNDS-LOVE#33]
# Description
The window for vnds opens, but I just get a blue screen with white error text. Let me know if I'm missing something obvious, or if you want me to test anything (my Mac is ready and willing for testing haha)

## Reproduce
1. Unzip vnds and love.
2. Run vnds (I'm running it from Applications)

## Screenshot
<img width="912" alt="Screen Shot 2021-03-06 at 5 44 49 PM" src="https://user-images.githubusercontent.com/3210509/110222883-a81a5880-7ea3-11eb-8518-1cffe4aa7df5.png">

## Environment
 - OS/Machine: macOS Catalina 10.15.7
 - VNDS-LOVE 0.0.7

### Error info
```
Error

text.lua:99: attempt to index upvalue 'text_font' (a nil value)


Traceback

text.lua:99: in function 'word_wrap'
text.lua:46: in function 'callback'
lib/event.lua:65: in function 'dispatch'
main.lua:159: in function 'load'
[C]: in function 'xpcall'
[C]: in function 'xpcall'
```

**Comment by ajusa** [comment]
Hm, that is strange. I assume that you aren't even opening a novel, correct?
Could you test https://github.com/ajusa/VNDS-LOVE/releases/tag/0.0.9 (pre-release version) and see if it works there? Also thanks for testing on a Mac, I don't have access to one so this is really helpful!

**Comment by lochlainn** [comment]
Ok, I got the exact same error result with the prerelease Mac binary. However, running the 0.09 .love file in LOVE.zip worked fine (I have the latest love binary in my Applications already).

I tried some simple bisecting by testing the Mac build for 0.08, and I got the expected result (with no novel as well):
<img width="912" alt="Screen Shot 2021-03-06 at 6 01 18 PM" src="https://user-images.githubusercontent.com/3210509/110224014-0d6f4900-7ea6-11eb-9a0a-0b5273700e54.png">

**Comment by ajusa** [comment]
Hm, so if I understood you correctly:
The mac build (binary) for 0.08 works, and the love file works for 0.08 and 0.09, but the binary doesn't work for 0.09? So in other words something messed up the binary built version between the two?

**Comment by lochlainn** [comment]
Yeah, that's the results I got. Are there any commits in particular you think would be worth trying to see where the bug happened?

**Comment by ajusa** [comment]
Hm, that is super weird. As far as I can tell, there should be no difference between the .love file and the mac binary for 0.09, so nothing should have broken. Having said that, the bug you are having is due to `text_font` not having loaded in. I'll push a commit that initializes it to the default font at least, and hopefully that'll fix the issue.

**Comment by ajusa** [comment]
Try https://github.com/ajusa/VNDS-LOVE/releases/tag/0.0.10 and see if that helps!

**Comment by lochlainn** [comment]
Working as expected now! Well, kinda. I realized that it keeps telling me to add a novel:
<img width="912" alt="Screen Shot 2021-03-06 at 6 24 05 PM" src="https://user-images.githubusercontent.com/3210509/110224387-334a1d00-7ea9-11eb-827d-85dbef1f0d51.png">

However, again trying the newest .love file, it finds the novel successfully:
<img width="912" alt="Screen Shot 2021-03-06 at 6 25 14 PM" src="https://user-images.githubusercontent.com/3210509/110224413-5ecd0780-7ea9-11eb-843d-811bed2c5171.png">

Should I be using a different novel directory for the Mac binary version?

**Comment by ajusa** [comment]
What does it say for the top screenshot? It should print out a path for the novel itself. For example, on my machine it says this when it can't find any novels:
![image](https://user-images.githubusercontent.com/7073362/110224665-7b6a3f00-7eab-11eb-82ee-6b4e52d7f071.png)

**Comment by SanktyRG** [comment]
Hello there, I just wanted to comment on this issue as the error that was initially occurring on the quoted post hast started appearing on Nintendo Switch with the new 0.0.10 pre-release. It's worth noting this only happens when there are novels in the folder it needs. When novels folder is empty it just shows the "No novels found in this directory:" message. Hope this report helps in any way.

> # Description
> The window for vnds opens, but I just get a blue screen with white error text. Let me know if I'm missing something obvious, or if you want me to test anything (my Mac is ready and willing for testing haha)
> 
> ## Reproduce
> Steps to reproduce the behavior:
> 
> 1. Unzip vnds and love.
> 2. Run vnds (I'm running it from Applications)
> 
> ## Screenshot
> <img alt="Screen Shot 2021-03-06 at 5 44 49 PM" width="912" src="https://user-images.githubusercontent.com/3210509/110222883-a81a5880-7ea3-11eb-8518-1cffe4aa7df5.png">
> 
> ## Environment
> * OS/Machine: macOS Catalina 10.15.7
> * VNDS-LOVE 0.0.7
> 
> ### Error info
> ```
> Error
> 
> text.lua:99: attempt to index upvalue 'text_font' (a nil value)
> 
> 
> Traceback
> 
> text.lua:99: in function 'word_wrap'
> text.lua:46: in function 'callback'
> lib/event.lua:65: in function 'dispatch'
> main.lua:159: in function 'load'
> [C]: in function 'xpcall'
> [C]: in function 'xpcall'
> ```

**Comment by lochlainn** [comment]
I haven’t gotten the chance to dig into this deeper, but those screenshots
I had were the full image; i.e. there was no directory printout. If you
have any suggestions, I’ll be on in the evening to try things.

**Comment by ajusa** [comment]
> Hello there, I just wanted to comment on this issue as the error that was initially occurring on the quoted post hast started appearing on Nintendo Switch with the new 0.0.10 pre-release. It's worth noting this only happens when there are novels in the folder it needs. When novels folder is empty it just shows the "No novels found in this directory:" message. Hope this report helps in any way.

Are you sure it is the exact same error as the one that was posted? Like the "text_font" upvalue being nil bit? Or is it a different error?

> I haven’t gotten the chance to dig into this deeper, but those screenshots
> I had were the full image; i.e. there was no directory printout. If you
> have any suggestions, I’ll be on in the evening to try things.

So that is super weird, there should be a directory printout (unless the Love2D stuff is error due to some sort of weird permission on OSX. I'll let you know once I have something for you to try.

**Comment by SanktyRG** [comment]
> > Hello there, I just wanted to comment on this issue as the error that was initially occurring on the quoted post hast started appearing on Nintendo Switch with the new 0.0.10 pre-release. It's worth noting this only happens when there are novels in the folder it needs. When novels folder is empty it just shows the "No novels found in this directory:" message. Hope this report helps in any way.
> 
> Are you sure it is the exact same error as the one that was posted? Like the "text_font" upvalue being nil bit? Or is it a different error?
> 
> > I haven’t gotten the chance to dig into this deeper, but those screenshots
> > I had were the full image; i.e. there was no directory printout. If you
> > have any suggestions, I’ll be on in the evening to try things.
> 
> So that is super weird, there should be a directory printout (unless the Love2D stuff is error due to some sort of weird permission on OSX. I'll let you know once I have something for you to try.

I'm really sorry, you are right. The error I'm getting is: "listbox.lua:11: attemp to call field 'newText' (a nil value)"

**Comment by drewbitt** [comment]
Yes, when a game folder is present in 0.10 on the Switch it has the error

```
listbox.lua:11: attempt to call field 'newText' (a nil value)
Traceback: 
listbox.lua:11 in function <listbox.lua:3>
```

**Comment by ajusa** [comment]
Alright, thanks for confirming the issue. I'll take a look when I get a chance!
EDIT: Actually, this is probably because newText isn't implemented on the Switch with the version of LovePotion that is being used. I still need to fix it myself, but it'll depend on #30. Closing this issue as I believe that the original problem on MacOS is solved. I've added a note to #30 for the newText bit.

**Comment by drewbitt** [comment]
0.0.11 works for starting on Switch. Thanks
